### DISCUSSION TITLE
Replace wpa2 with Wpa3-SAE?

### DISCUSSION THREAD

**Discussion by ghost** [radio24/TorBox#130] [Development]
Wpa2 to easy to crack with simple gpu. And it is very easy to decrypt WPA2-psk traffic. 

Love the project but think that this is the biggest weakness.

**Comment by nyxnor** [comment]
not every device will have wpa3 support, maybe just give an option.
I found this SAE [config](https://dokuwiki.alu4u.com/doku.php?id=stellar-wireless-wpa3-sae-raspberry-pi)
```
ctrl_interface=DIR=/var/run/wpa_supplicant GROUP=netdev
eapol_version=1
ap_scan=1
fast_reauth=1
country=DE
pmf=1
network={
	ssid="ov44b39"
	proto=RSN
	key_mgmt=SAE
	sae_password="12345678"
}
```

**Reply by radio24** [reply]
I guess he means to create a WPA3 AP (TorBox's wifi), which make sense. On the other hand, there is no security risk if the wifi connection between the TorBox and a WPA2 hotspot is cracked because there is only Tor data traffic from the clients, which is itself encrypted. This is why TorBox can be used to connect insecure open wifis.

**Comment by radio24** [comment]
> Wpa2 to easy to crack with simple gpu. And it is very easy to decrypt WPA2-psk traffic.
> 
> Love the project but think that this is the biggest weakness.

The TorBox Wireless Manager uses wpa_supplicant to connect a wifi network. I guess it should work with WPA3, but I didn't have the chance to verify/test it. Anyway, connecting to an insecure wireless network shouldn't be a point of weakness (see explanation above).

To use WPA3 for the TorBox wifi is another story, and I put your idea on my to-do list. However, so far, it seems that the WPA3 support is not very good yet.

**Reply by ghost** [reply]
super yes i meant wpa3 sae AP. (To block man in the middle attack) i know once traffic leaves torbox it is encrypted. I want to donate €50 to the project if wpa3 sae AP works.

**Comment by radio24** [comment]
**Update**
While preparing to release TorBox v.0.5.3, I tested several options to enable WPA3 on the Raspberry Pi's AP, but I wasn't successful. Seeing on Internet help sites and forums all the issues people have with WPA3 on a Raspberry Pi, it seems that it still doesn't support WPA3. Nevertheless, I will continue to follow this topic.

**Reply by ghost** [reply]
thanks for all the work you put in..
Love the project!

**Comment by radio24** [comment]
In preparation for the next TorBox version, I looked into this issue again and tried to create a WPA3-SAE Access Point (AP).

For a WPA3-SAE AP under Linux using `hostapd`, a few prerequisites are necessary:
- WPA3-SAE is only supported from `hostapd` 2.7 or newer and has to be compiled with SAE support.
- AP side: Compatible WLAN hardware and kernel drivers (e.g. `nl80211`) are needed.
- Client side: The clients that are to connect must support WPA3-SAE.

Regarding the configuration, `hostapd` has to be added with the following lines:
- `wpa_key_mgmt=SAE` for WPA3-SAE only or `wpa_key_mgmt=SAE WPA-PSK` for WPA2/WPA3 mixed mode.
- `ieee80211w=2` for WPA3-SAE only or `ieee80211w=1` for WPA2/WPA3 mixed mode.

The `hostapd` package from the official Raspberry Pi OS repositories is up to date and supports WPA3-SAE. However, clients cannot connect to the AP using the onboard Wi-Fi on most Raspberry Pi models (3, 4, 5). The issue appears to be related to the incorrect password, but unfortunately, the problem seems to lie in the [chipset of the onboard Broadcom-Chip](https://docs.raspap.com/ap-basics/)/firmware/driver. This situation is independent of the Raspberry Pi model, the firmware and the Linux kernel.

Also, all distributions seem to face identical limitations because they all use:
- same hardware constraints
- same Broadcom/Cypress firmware blobs
- same brcmfmac kernel driver

**Possible workaround**
WPA3 support for an AP may be possible with some WiFi USB-adapters. However, we had a very mixed experience using WiFi USB-adapters for AP, even with WPA2. Therefore, we believe this solution is only suitable for experts and shouldn't be integrated into the TorBox menus at this time.
